scripts/format.sh

```diff
@@ -36,9 +36,9 @@ formatter=clang-format-9
 
 if [ $# -eq 0 ]; then
     cd "$(git rev-parse --show-toplevel)"
-    find src/ include/ test/ examples/ -print0 \
-        -type f \( -name '*.h' -o -name '*.cpp' \) \
-    | xargs "$formatter" -i --style=file
+    find src/ include/ test/ examples/ \
+        -type f \( -name '*.h' -o -name '*.cpp' \) -print0 \
+    | xargs --null "$formatter" -i --style=file
```

Comment:
I think `--null` is a gnu-ism, and `-0` (zero) is more common, or using find's `-exec` flag to avoid xargs entirely.

Reply:
I like how `xargs` allows us to run the formatter only once. It's not a performance argument, just something that I like.  Will change to `-0`.